slack-java-client/src/test/java/com/hubspot/slack/client/SlackWebClientTest.java

```diff
@@ -0,0 +1,35 @@
+package com.hubspot.slack.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.hubspot.algebra.Result;
+import com.hubspot.horizon.HttpRequest;
+import com.hubspot.slack.client.methods.SlackMethods;
+import com.hubspot.slack.client.models.response.SlackError;
+import com.hubspot.slack.client.models.response.SlackErrorType;
+import com.hubspot.slack.client.models.response.SlackResponse;
+
+public class SlackWebClientTest {
+
+  @Test
+  public void itProactivelyRateLimits() {
+    SlackWebClient slackClient = (SlackWebClient) SlackClientFactory.defaultFactory().create(SlackClientRuntimeConfig.builder()
+            .setTokenSupplier(() -> "xoxp-test-token")
+            .setRequestDebugger((requestId, method, request) -> {})
+            .setSlackRateLimiter((slackToken, slackMethod) -> SlackWebClient.RATE_LIMIT_SENTINEL_VALUE)
```

Comment:
This isn't meant to be a thorough test for sending requests. It's just a sanity check to ensure we have something checking for the rate limit sentinel value